### PR TITLE
fix(dropdown): dark mode broken

### PR DESCRIPTION
**Pull request by EmelieLitwin**
**Describe pull-request**  
Dark mode didn't work. New PR on my old branch.

**Solving issue**  
Fixes: [AB#3325](https://dev.azure.com/scaniadigitaldesignsystem/3d9eb9c6-0045-473d-bc1d-e842dd779200/_workitems/edit/3325)

### CODE REVIEW

File: tegel/src/components/dropdown/dropdown-theme-vars.scss [hunk 1]
```diff
@@ -4,15 +4,39 @@
...
+  --sdds-dropdown-background: var(--sdds-white);
+  --sdds-dropdown-background-disabled: var(--sdds-white);
+  --sdds-dropdown-disabled: var(--sdds-grey-400);
+  --sdds-dropdown-background-hover: var(--sdds-grey-50);
+  --sdds-dropdown-border: var(--sdds-grey-100);
+  --sdds-dropdown-label-inside: var(--sdds-grey-700);
+  --sdds-dropdown-label-outside: var(--sdds-grey-958);
+  --sdds-dropdown-placeholder: var(--sdds-grey-700);
+
+  .sdds-on-white-background {
```
Comment (nathalielindqvist) [inline tegel/src/components/dropdown/dropdown-theme-vars.scss:18]: Do we still need this?

File: tegel/src/components/dropdown/dropdown-theme-vars.scss [hunk 2]
```diff
@@ -4,15 +4,39 @@
...
+  --sdds-dropdown-background-hover: var(--sdds-grey-800);
+  --sdds-dropdown-border: var(--sdds-grey-800);
+  --sdds-dropdown-label-inside: var(--sdds-grey-400);
+  --sdds-dropdown-label-outside: var(--sdds-grey-100);
+  --sdds-dropdown-placeholder: var(--sdds-grey-400);
 
-  .sdds-on-white-bg {
-    --sdds-dropdown-bg: var(--sdds-grey-50);
-    --sdds-dropdown-bg-hover: var(--sdds-grey-100);
+  .sdds-on-white-background {
```
Comment (nathalielindqvist) [inline tegel/src/components/dropdown/dropdown-theme-vars.scss:38]: Do we still need this? If yes, is it supposed to say something else than *white* background since it's under dark theme variables?

File: tegel/src/components/dropdown/dropdown-theme-vars.scss [hunk 3]
```diff
@@ -4,15 +4,39 @@
...
+.sdds-theme-dark {
+  --sdds-link: var(--sdds-grey-400);
   --sdds-link-text-decoration: none;
-  --sdds-dropdown-bg: var(--sdds-white);
-  --sdds-dropdown-bg-hover: var(--sdds-grey-50);
+  --sdds-dropdown-background: var(--sdds-grey-900);
+  --sdds-dropdown-background-disabled: var(--sdds-grey-900);
+  --sdds-dropdown-disabled: var(--sdds-grey-800);
+  --sdds-dropdown-background-hover: var(--sdds-grey-800);
+  --sdds-dropdown-border: var(--sdds-grey-800);
```
Comment (nathalielindqvist) [inline tegel/src/components/dropdown/dropdown-theme-vars.scss:33]: Seems like the dark mode border is Grey600 in Figma, har this changed? :)
image

File: tegel/src/components/dropdown/dropdown.scss [hunk 4]
```diff
@@ -140,18 +140,19 @@
...
     min-width: 100%;
     max-height: 360px; // max 7 option items, then add scroll
     border-color: transparent;
     border-style: solid;
-    border-width: 0 1px;
+    border-width: 0 1px 1px;
     box-shadow: 0 2px 3px 0 rgb(0 0 0 / 10%);
-    border-left-color: var(--sdds-grey-100);
-    border-right-color: var(--sdds-grey-100);
+    border-left-color: var(--sdds-dropdown-border);
```
Comment (nathalielindqvist) [inline tegel/src/components/dropdown/dropdown.scss:153]: Might not be part of this PR, but there seems to be instances where the border is still set with the generic color variable, not with tokens like here. Should that be changed? Couldn't find in which file(s) it's applied
image

It also seems to overwrite it's own border color in the native component, should we change that?
image

**Review by nathalielindqvist — COMMENTED** [review]
Awesome that you found what caused the dark mode not to work! 🎊 
Left a few comment on stuff I'm unsure about, let me know if they're not applicable here :relaxed:

File: tegel/src/components/dropdown/dropdown-theme-vars.scss (hunk 1 above)
Comment (mJarsater) [reply tegel/src/components/dropdown/dropdown-theme-vars.scss:18]: Njiet, I'll remove it 👍🏻

File: tegel/src/components/dropdown/dropdown-theme-vars.scss [hunk 5]
```diff
@@ -1,44 +1,56 @@
...
+  --sdds-dropdown-background: var(--sdds-grey-868);
+  --sdds-dropdown-background-hover: var(--sdds-grey-800);
   --sdds-dropdown-background-disabled: var(--sdds-grey-900);
   --sdds-dropdown-disabled: var(--sdds-grey-800);
-  --sdds-dropdown-background-hover: var(--sdds-grey-800);
   --sdds-dropdown-border: var(--sdds-grey-800);
   --sdds-dropdown-label-inside: var(--sdds-grey-400);
   --sdds-dropdown-label-outside: var(--sdds-grey-100);
   --sdds-dropdown-placeholder: var(--sdds-grey-400);
+  --sdds-dropdown-border-not-focused: var(--sdds-grey-600);
```
[the remark this answers is not in the document]
Comment (mJarsater) [reply tegel/src/components/dropdown/dropdown-theme-vars.scss:43]: Yes! Will do :)

File: tegel/src/components/dropdown/_dropdown-core.scss [hunk 6]
```diff
@@ -3,15 +3,15 @@
...
+  color: var(--sdds-dropdown-color);
   display: flex;
   align-items: center;
   background-color: var(--sdds-dropdown-background);
   cursor: pointer;
   transition: box-shadow 0.1s ease-in, border-bottom-color 150ms ease;
   border-radius: 4px 4px 0 0;
   border: none;
-  border-bottom: 1px solid var(--sdds-grey-400);
+  border-bottom: 1px solid var(--sdds-dropdown-border-not-focused);
```
Comment (nathalielindqvist) [inline tegel/src/components/dropdown/_dropdown-core.scss:14]: Seems like this is still getting overwritten so the border is still Grey800 - is there a way around this? :)
image

Comment (mJarsater) [reply tegel/src/components/dropdown/_dropdown-core.scss:14]: Oops, my bad. Pushed correction now! :)

**Review by nathalielindqvist — APPROVED** [review]
Looking good! 🔥
